packages/widget-v2/src/widget/Router.tsx

```diff
@@ -0,0 +1,31 @@
+import { ErrorPage } from '@/pages/ErrorPage/ErrorPage';
+import { SwapExecutionPage } from '@/pages/SwapExecutionPage/SwapExecutionPage';
+import { SwapPage } from '@/pages/SwapPage/SwapPage';
+import { errorAtom } from '@/state/errorPage';
+import { Routes, currentPageAtom } from '@/state/router';
+import { useAtom } from 'jotai';
+import { ErrorBoundary } from 'react-error-boundary';
+
+export const Router = () => {
+  const [currentPage] = useAtom(currentPageAtom);
+  const [error, setError] = useAtom(errorAtom);
+
+  if (error) {
+    return <ErrorPage />;
+  }
+
+  switch (currentPage) {
+    case Routes.SwapPage:
+      return (
+        <ErrorBoundary fallback={null} onError={(error) => setError(error)}>
+          <SwapPage />
+        </ErrorBoundary>
+      );
+    case Routes.SwapExecutionPage:
+      return (
+        <ErrorBoundary fallback={null} onError={(error) => setError(error)}>
+          <SwapExecutionPage />
+        </ErrorBoundary>
+      );
+  }
+};
```

Comment:
I tested manually throwing an error within SwapPage and it successfully sets errorAtom state and renders ErrorPage

Reply:
I also tested and can confirm that these error boundaries **will not** catch any errors thrown within a modal, we will need to handles those seperately